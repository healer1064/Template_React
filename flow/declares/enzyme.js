// @flow

declare module 'enzyme' {

  declare var shallow: Function

}
