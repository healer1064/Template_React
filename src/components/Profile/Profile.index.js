// @flow
import { connect } from 'react-redux'
import Profile from './Profile'


type ProfileStateProps = {
}


function mapStateToProps (state: GlobalReducerState): ProfileStateProps {

  const { init } = state

  return {
    init,
  }

}


export default connect(mapStateToProps)(Profile)
