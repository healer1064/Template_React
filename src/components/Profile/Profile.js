// @flow
import React from 'react'
import { Link } from 'react-router'
import css from './Profile.style.css'


import type { ReduxProps } from './Profile.index'

type Props = ReduxProps


const Profile = (props: Props): React$Element<any> => {

  const { me } = props

  return (
    <div className={`${css.profile} pt7 tc`}>
      Profile page
      <ul>
        <li>{ me.get('city') }</li>
      </ul>
      <Link to="/">Back</Link>
    </div>
  )

}


export default Profile
