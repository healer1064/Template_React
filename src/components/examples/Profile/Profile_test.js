// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import Profile from './Profile'

type OverrideProps = {
  me: Object,
  error: string,
  location: Object,
  history: Object,
  fetchProfile: Function,
}

const mockProps: OverrideProps = {
  me: {
    get: () => null,
  },
  error: '',
  location: {},
  history: {
    replace: () => {},
  },
  fetchProfile: () => {},
}

it('<Profile> renders the Profile content', () => {

  const wrapper = shallow(
    <Profile
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
