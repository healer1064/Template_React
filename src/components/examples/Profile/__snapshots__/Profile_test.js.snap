// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> displays profile info 1`] = `
<div
  class="profile"
>
  <div
    class="content"
  >
    Profile page
    <ul>
      <li />
      <li />
      <li>
        Austin, TX
      </li>
    </ul>
    <a>
      Back
    </a>
    <button>
      Refresh
    </button>
    <div
      class="error"
    />
  </div>
</div>
`;
